cmake/toolchain/cross-compile/generic.cmake

```diff
@@ -28,4 +28,6 @@ set(COMPILER gcc)
 set(LINKER ld)
 set(BINTOOLS gnu)
 
+set(TOOLCHAIN_HAS_NEWLIB ON CACHE BOOL "True if toolchain supports newlib")
```

Comment:
The problem with this line is that we can not know if the `cross-compile` toolchain actually has newlib support built-in or not.

One option is to actually tell Zephyr CMake that the toolchain does support newlib, by adding `-DTOOLCHAIN_HAS_NEWLIB=y` when invoking CMake.

Reply:
Thank you for your response. I think before 3.7 it used to work and a lot of linux distribution (at least Gentoo and Ubuntu) have the arm-none-eabi built with newlib. So trying to compile zephyr with this toolchain results in the error message:

`arm-none-eabi-gcc: fatal error: cannot read spec file 'picolibc.specs': No such file or directory`

To add the `-DTOOLCHAIN_HAS_NEWLIB=y` as a CMake argument is not the most obvious solution.
Like `CROSS_COMPILE` and `ZEPHYR_TOOLCHAIN_VARIANT` I think it should be an environment variable and mentioned in the "Other Cross Compilers" documentation.

Or maybe it is possible that cmake can detect the correct libc variant?

Reply:
ref: https://docs.zephyrproject.org/latest/develop/toolchains/gnu_arm_embedded.html